test/functional/wallet_balance.py

```diff
@@ -257,5 +259,33 @@ def test_balances(*, fee_node_1=0):
         assert_equal(self.nodes[0].getbalance(minconf=0), total_amount + 1)  # The reorg recovered our fee of 1 coin
 
 
+        #Tests the lastprocessedblock JSON object in getbalances, getwalletinfo
+        #and gettransaction by checking for valid hex strings and by comparing
+        #the hashes & heights between generated blocks.
+        self.log.info("Test getbalances returns expected lastprocessedblock json object")
+        self.nodes[0].generatetoaddress(2, self.nodes[0].get_deterministic_priv_key().address)
+        balances = self.nodes[0].getbalances()
+        assert_greater_than(balances['mine']['immature'], 0)
+
+        prev_hash = self.nodes[0].getbestblockhash()
+        prev_height = self.nodes[0].getblock(prev_hash)['height']
+        self.nodes[0].generatetoaddress(5, self.nodes[0].get_deterministic_priv_key().address)
+        lastblock = self.nodes[0].getbalances()['lastprocessedblock']
+        assert_is_hash_string(lastblock['hash'])
+        assert_equal((prev_hash == lastblock['hash']), False)
+        assert_greater_than(lastblock['height'], prev_height)
+
+        self.log.info("Test getwalletinfo returns expected lastprocessedblock json object")
+        walletinfo = self.nodes[0].getwalletinfo()
+        assert_greater_than(walletinfo['lastprocessedblock']['height'], 0)
+        assert_is_hash_string(walletinfo['lastprocessedblock']['hash'])
+
+        self.log.info("Test gettransaction returns expected lastprocessedblock json object")
+        txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 1)
+        self.nodes[0].generatetoaddress(6, self.nodes[0].get_deterministic_priv_key().address)
+        tx_info = self.nodes[0].gettransaction(txid)
+        assert_greater_than(walletinfo['lastprocessedblock']['height'], 0)
```

Comment:
I think you want to access the just one line above returned `tx_info` here, not `walletinfo`?

Reply:
Thatnks!
Yes, `walletinfo` in this part is redundant as it got already checked in the previous code block.